### PR TITLE
[SPARK-8382][SQL] Improve Analysis Unit test framework

### PR DESCRIPTION
**Pull request by cloud-fan**

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TestRelations.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.logical.LocalRelation
+import org.apache.spark.sql.types._
+
+object TestRelations {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TestRelations.scala]: why is this file in main rather than source?

Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TestRelations.scala]: what does the `main` and `source` mean?

**Comment by cloud-fan** [comment]
retest this please.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TestRelations.scala (hunk 1 above)
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TestRelations.scala]: I meant main vs test.

**Comment by marmbrus** [comment]
Thanks!  Merged to master.

**Comment by marmbrus** [comment]
and 1.5

**Pull request closed without merging**